master/internal/proxy/proxy_intg_test.go

```diff
@@ -0,0 +1,101 @@
+package proxy
+
+import (
+	"net/url"
+	"testing"
+
+	"github.com/labstack/echo/v4"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	proxyAuth = func(c echo.Context) (done bool, err error) {
+		return true, nil
+	}
+	serviceIDs = []string{"a", "b", "c"}
+	u          = url.URL{} // "localhost:8081" TODO CAROLINA
+)
+
+func register(t *testing.T, prTCP bool, unauth bool) {
+	for _, id := range serviceIDs {
+		DefaultProxy.Register(id, &u, prTCP, unauth)
+		if DefaultProxy.GetService(id) == nil {
+			t.Logf("failed to find registered service %s", id)
+		}
+	}
+	if len(DefaultProxy.Summary()) != len(serviceIDs) {
+		t.Logf("failed to register all services")
+	}
+}
+
+func unregister(t *testing.T) {
+	for _, id := range serviceIDs {
+		if DefaultProxy.GetService(id) == nil {
+			t.Logf("failed to find registered service %s", id)
```

Comment:
in all the places you used `t.Log` to log failures you probably should be using `t.Error` instead

Reply:
there are a few redundant checks from this, i think

Reply:
what do you mean redundant?

Reply:
ooh got it now -- removed 
```
	if len(DefaultProxy.Summary()) != 0 {
		t.Errorf("failed to unregister all services.")
	}
```